File: ios/src/main/java/com/vfpowertech/jsbridge/ios/MessageHandler.java
package com.vfpowertech.jsbridge.ios;

import org.robovm.apple.foundation.Foundation;
import org.robovm.apple.foundation.NSObject;
import org.robovm.apple.webkit.WKScriptMessage;
import org.robovm.apple.webkit.WKScriptMessageHandlerAdapter;
import org.robovm.apple.webkit.WKUserContentController;

public class MessageHandler extends WKScriptMessageHandlerAdapter {
    @Override
    public void didReceiveScriptMessage(WKUserContentController wkUserContentController, WKScriptMessage wkScriptMessage) {
        Foundation.log("Called");
        String name = wkScriptMessage.getName();
        NSObject body = wkScriptMessage.getBody();
        if (!name.equals("native")) {
            Foundation.log("Unsupported handler");
            return;
        }
        Foundation.log(String.format("Message from js: %s/%s", name, body));
        Foundation.log("Done");
    }
}

File: ios/src/main/kotlin/com/vfpowertech/jsbridge/ios/MessageHandler.kt
package com.vfpowertech.jsbridge.ios

import org.robovm.apple.foundation.Foundation
import org.robovm.apple.webkit.WKScriptMessage
import org.robovm.apple.webkit.WKScriptMessageHandlerAdapter
import org.robovm.apple.webkit.WKUserContentController

class MessageHandler : WKScriptMessageHandlerAdapter() {
    override fun didReceiveScriptMessage(wkUserContentController: WKUserContentController, wkScriptMessage: WKScriptMessage) {
        Foundation.log("Called")
        val name = wkScriptMessage.name
        val body = wkScriptMessage.body
        if (name != "native") {
            Foundation.log("Unsupported handler")
            return
        }
        Foundation.log("Message from js: %s/%s".format(name, body))
        Foundation.log("Done")
    }
}